stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -376,6 +376,27 @@ function SparseMatrixCSC{Tv,Ti}(M::AbstractMatrix) where {Tv,Ti}
     eltypeTvV = convert(Vector{Tv}, V)
     return sparse_IJ_sorted!(eltypeTiI, eltypeTiJ, eltypeTvV, size(M)...)
 end
+function SparseMatrixCSC{Tv,Ti}(M::Matrix) where {Tv,Ti}
+    nz = count(t -> t != 0, M)
+    colptr = zeros(Ti, size(M, 2) + 1)
+    nzval = Vector{Tv}(uninitialized, nz)
+    rowval = Vector{Ti}(uninitialized, nz)
+    colptr[1] = 1
+    cnt = 1
+    @inbounds for j in 1:size(M, 2)
+        for i in 1:size(M, 1)
+            v = M[i, j]
+            if v != 0
+                rowval[cnt] = i
+                nzval[cnt] = v
+                cnt += 1
+            end
+        end
+        colptr[j+1] = cnt
+    end
+    return SparseMatrixCSC(size(M, 1), size(M, 2), colptr, rowval, nzval)
+end
+
 # converting from SparseMatrixCSC to other matrix types
 function Matrix(S::SparseMatrixCSC{Tv}) where Tv
```

Comment:
Hi.

You are using

>  size(M, 2)

 3 times and 

> size(M,1)

 two times. Would not be faster to define two variables with those values and than use them? Or if the interpreter can optimize it anyway ?

Reply:
That's a valid comment. However, in this case, even if the compiler doesn't optimize it away, the time to access these fields should be completely insignificant to the rest of the code. The question here is if it makes the code more legible to store them in `m` and `n`, for example. I'm not sure.

Reply:
Indeed! Thanks for your answer. Sometimes I wonder if the loops can be (somehow) optimized in those cases.